presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -529,6 +527,18 @@ public void setOutput(Optional<Output> output)
         this.output.set(output);
     }
 
+    private void attachConnectorCommitHandleToOutput(ConnectorCommitHandle commitHandle)
+    {
+        if (output.get().isPresent()) {
+            Output outputInfo = output.get().get();
+            output.set(Optional.of(new Output(
+                    outputInfo.getConnectorId(),
+                    outputInfo.getSchema(),
+                    outputInfo.getTable(),
+                    commitHandle)));
```

Comment:
We don't need to pass the commit handle here. Instead, pass the `commitHandle.getCommitOutput()`. Also, use `JsonCodec<Object>` to serialize the object. Check `TableFinishInfo` as an example, it is how `ConnectorOutputMetadata.getInfo()` got serialized.

Reply:
Will check how it works.